### PR TITLE
Outfit wars support

### LINKED ISSUE #312: Outfitwars support
Support for https://github.com/ps2alerts/aggregator/issues/578 and https://github.com/ps2alerts/website/issues/460

Will require new endpoints or current ones adjusting to accept a type of metagame instance.

## Aggregator support

- We will need the ability to process aggregations, **but** they must now accept a type e.g. `type: outfitwarsterritory`.
- Endpoints will need to be adjusted to allow the website to query by a type of instance. By default it should serve `TerritoryInstance`, but optionally we should able to specify the instance type for `OutfitWarsTerritoryInstance`.

### PR DESCRIPTION
**Pull request by Maelstromeous**
Work towards adding support for outfit wars. Closes #312 

## Tasks

- [x] Create API endpoints to create OW instance
- [x] Create update endpoint to update result from aggregator
- [x] Test deleting (trashing) via aggregator
- [x] Create filters on **All Aggregates** to filter by Outfit Wars (global and instance)
- [x] Update all aggregates DTOs / entities to add `ps2alertsEventType`
- [x] Create database migration to add a `ps2alertsEventType` migration to ALL aggregates, giving all current records the value of `1` (live metagame)
- [ ] Create endpoints for website to hit for statistical data e.g. describing the current rounds etc (theoretical atm)
- [x] Add API to update outfit ranking with instance ID for linking rankings to matches. Will need to filter on round number and outfit ID
- [ ] Increase frequency of outfitwars ranking cron job 
- [x] Have the rankings cron job choose the round based on the number of matches played (round = matches played + 1)

### CODE REVIEW

File: src/modules/cron/population.history.cron.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
+// import {Inject, Injectable, Logger} from '@nestjs/common';
```
Comment (Maelstromeous) [inline src/modules/cron/population.history.cron.ts:1]: This file shouldn't exist in this PR, my bad

File: src/modules/data/entities/instance/outfitwars.mapcontrol.embed.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import {ApiProperty} from '@nestjs/swagger';
+import {Column} from 'typeorm';
+
+export default class OutfitwarsMapControlEmbed {
+    @ApiProperty({example: 20, description: 'Team 1 (blue) percentage'})
```
Comment (Maelstromeous) [inline src/modules/data/entities/instance/outfitwars.mapcontrol.embed.ts:5]: Old descriptions

File: src/modules/rest/Dto/UpdateFacilityControlDto.ts [hunk 3]
```diff
@@ -43,5 +43,5 @@ export class UpdateFacilityControlDto {
     @IsObject()
     @IsOptional()
     @ApiModelProperty({example: null, default: null})
-    mapControl: MapControlEmbed;
+    mapControl: TerritoryControlMapControlEmbed;
```
Comment (Maelstromeous) [inline src/modules/rest/Dto/UpdateFacilityControlDto.ts:46]: Thinking we should probably prefix this with metagame so it's clear is not OW related, like the rest are

File: src/modules/rest/Dto/outfitwars/UpdateFacilityControlOutfitWarsDto.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    @IsNotEmpty()
+    @IsOptional()
+    @ApiModelProperty({example: 1234})
+    durationHeld: number;
+
+    @IsBoolean()
+    @IsNotEmpty()
+    @IsOptional()
+    @ApiModelProperty({example: false, default: false})
+    isInitial: boolean;
```
Comment (Maelstromeous) [inline src/modules/rest/Dto/outfitwars/UpdateFacilityControlOutfitWarsDto.ts:36]: Potentially can remove

File: src/modules/rest/controllers/rest.outfitwars.controller.ts [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    constructor(
+        @Inject(MongoOperationsService) private readonly mongoOperationsService: MongoOperationsService,
+        private readonly cacheService: RedisCacheService,
+    ) {}
+
+    @Get('/:instance')
+    @ApiOperation({summary: 'Returns a single outfit wars instance'})
+    @ApiResponse({
+        status: 200,
+        description: 'The Metagame Instance',
```
Comment (Maelstromeous) [inline src/modules/rest/controllers/rest.outfitwars.controller.ts:67]: Wonky description

File: src/modules/rest/controllers/rest.outfitwars.controller.ts [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+    @ApiUnauthorizedResponse({description: 'This is an internal PS2Alerts endpoint, you won\'t have access to this - ever.'})
+    @ApiSecurity('basic')
+    @UseGuards(AuthGuard('basic'))
+    async deleteOne(
+        @Param('instance') instanceId: string,
+    ): Promise<boolean> {
+        return await this.mongoOperationsService.deleteOne(InstanceOutfitWarsTerritoryEntity, {instanceId});
+    }
+
+    @Get('/active')
```
Comment (Maelstromeous) [inline src/modules/rest/controllers/rest.outfitwars.controller.ts:119]: Needs testing

File: src/modules/rest/controllers/rest.outfitwars.controller.ts [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+        return await this.cacheService.get(key) ?? await this.cacheService.set(
+            key,
+            await this.mongoOperationsService.findMany(
+                InstanceOutfitWarsTerritoryEntity,
+                {
+                    state: Ps2alertsEventState.STARTED,
+                    world,
+                    zone,
```
Comment (Maelstromeous) [inline src/modules/rest/controllers/rest.outfitwars.controller.ts:146]: Zone may not be needed here

**Review by Maelstromeous — COMMENTED** [review]
Various notes

**Comment by ryanjsims** [comment]
- [ ] Add API to update outfit ranking with instance ID for linking rankings to matches. Will need to filter on round number and outfit ID

**Pull request merged**